content/How to work with Stripe.md

```diff
@@ -0,0 +1,133 @@
+
+In that model you will store attribures like: `expires_at`, `type` or even `provider` if later you want to extend to other providres like PayPal or Apple Pay
+
+Finally to create them on Stripe is quite simple:
+
+https://gist.github.com/fedeagripa/d49a024299d2b8e315ff0861d32e9534
+
+### COUPONS
+Coupons are the abstract concept of `30% off` for example, when you apply that coupon to a user that's called a `discount`.
+So you should define some discounts on Stripe and store their ids in your database to apply them to users.
+There are two types of coupons `percentage` & `fixed amount`, and any of them can be one time only or have the capability to be applied multiple times. So when you try to apply a coupon to a subscription for example, remember that it can fail if you reached the maximum usage number.
+
+Another useful case that is worth mentioning is to apply a coupon to a user, this means that they will have a positive balance for any future invoice (be careful if you charge users with multiple products)
+
+### SUBSCRIPTION ITEMS
+These are your billing items, so for the case of a web subscription you will just have 1 subscription item. For specific cases like an amazon cart or any complicated use case (where you have multiple items being added to a purchase) is where you have to start considering adding some specific logic to your app.
+I won't get really into detail about this, I just wanted to show the general concept behind this, maybe I will write more in detail in a future post.
+
+##### RENEWALS
+Don't overthink it, there is a webhook for most of your use cases. But for this specific need you can configure the following events:
+
+ - customer.subscription.updated
+This event happens every time a susbscription is updated according to this [documentation](https://stripe.com/docs/billing/subscriptions/change)
+
+ - customer.subscription.deleted
+As simple as it sounds, it tells you when a subscription is canceled so you can take the actions needed in your app (possibly dissable the associated account)
+
+ - invoice.payment_succeeded
+This is a really important one! It tells us when a payment is actually accepted by the credit card provider (some times they can be fraud or declined)
+
+### WEBHOOKS
+There are a lot of them and they will solve most of your problems, the only downcase is the headache trying to understand which exactly to use.
+I'm sorry to dissapoint you if you reached here trying to answer this question but up to now I only know [this page](https://stripe.com/docs/api/events/types) that explains the different existing webhooks and what they do. The other option is when you go to create a webhook from the developer's Stripe dashboard, they explain a bit more in detail what each event does.
+
+## 4) SPECIAL RECOMMENDATIONS
+Keep these Stripe documentation pages as your friends:
+ - [Devs api](https://stripe.com/docs/api/)
+ - [Events types](https://stripe.com/docs/api/events/types)
```

Comment:
You're sharing the same page here and above (line 120). Just in case this wasn't intended

Reply:
It was, as stripe doc is a mess and when you cmd + F they launch a custom search I decided to shared different sections besides being the same page